pkg/policy/l4.go

```diff
@@ -59,53 +149,20 @@ func CreateL4Filter(rule api.PortRule, port api.PortProtocol, direction string,
 	}
 
 	if rule.Rules != nil {
-		l7rules := []AuxRule{}
-		for _, h := range rule.Rules.HTTP {
-			r := AuxRule{}
-
-			if h.Path != "" {
-				r.Expr = "PathRegexp(\"" + h.Path + "\")"
-			}
-
-			if h.Method != "" {
-				if r.Expr != "" {
-					r.Expr += " && "
-				}
-				r.Expr += "MethodRegexp(\"" + h.Method + "\")"
-			}
-
-			if h.Host != "" {
-				if r.Expr != "" {
-					r.Expr += " && "
-				}
-				r.Expr += "HostRegexp(\"" + h.Host + "\")"
+		switch {
+		case rule.Rules.HTTP != nil:
+			l7rules := FillPortRuleHTTP(rule.Rules.HTTP)
```

Comment:
You can consider having `FIllPortRuleHTTP` and `FillPortRuleKafka` return both the rules and the parser type so you can write:
```
l4.L7Rules, l4.L7Parser = FillPortRuleHTTP(rule.Rules.HTTP)
```

Reply:
sure

Reply:
done